backends/core/models.py

```diff
@@ -74,6 +83,21 @@ def recipients(self):
       return self.emails_for_notifications.split()
     return []
 
+  def _add_multi_to_memcache(self, mapping,retries = 10, time = MEMCACHE_DEFAULT_EXPIRATION_TIME):
+    """
+        Set multiple values in memcache, prefixed by the id of the pipeline
+        and having a default expiration time of 24 hours
+    """
+    client = memcache.Client()
+    while retries > 0:
+      cached_mapping = client.get_multi(mapping, key_prefix = str(self.id) + "_", for_cas = True)
```

Comment:
Can you put the building of the "key prefix" into a separate method? Since it's used in multiple places.

Reply:
Also, avoid concatenating string with `+` in Python. Prefer the string interpolation like this:

```python
key_prefix = '%s_' % str(self.id)
```